tests/compile-fail/non_extern_foreign_call/abi_mismatch.rs

```diff
@@ -0,0 +1,10 @@
+#[no_mangle]
+fn foo() {}
+
+fn main() {
+    extern "C" {
+        fn foo();
+    }
+    unsafe { foo() }
+    //~^ ERROR Undefined Behavior: calling a function with ABI Rust using caller ABI C
+}
```

Comment:
Given there are just two tests I am not sure if it is worth having a new folder for this.